plugins/automation/lib/discourse_automation/triggers/topic.rb

```diff
@@ -12,17 +12,19 @@
         previous_topic = Topic.find_by(id: previous_topic_id)
 
         if previous_topic
-          TopicCustomField.where(
-            topic_id: previous_topic_id,
-            name: DiscourseAutomation::CUSTOM_FIELD,
-            value: automation.id,
-          ).delete_all
+          automation.remove_id_from_custom_field(
+            previous_topic,
+            DiscourseAutomation::AUTOMATION_IDS_CUSTOM_FIELD,
+          )
         end
       end
 
       if topic_id
         topic = Topic.find_by(id: topic_id)
-        topic&.upsert_custom_fields(DiscourseAutomation::CUSTOM_FIELD => automation.id)
```

Comment:
Here's another bug that will fixed in this PR. The usage of `Topic#upsert_custom_fields` on a custom field of type array is incorrect because the method finds an existing record and updates its value which means the custom field will always be a single-value array. The right thing here would've been to use `Automion#attach_custom_field` which creates a new record for every new value in the array, but with the migration to JSON type, the right thing now is to maintain a single record that holds multiple values.